File: valid-parentheses/valid-parentheses.go
func isValid(s string) bool {
    stack := []byte{}
    
    for i:= 0; i<len(s); i++{
        switch s[i]{
            case '(','{','[':
                stack = append(stack,s[i])
            case ')','}',']':
            if len(stack) == 0{
                return false
            }
            
            lastChar := stack[len(stack)-1]
            stack = stack[:len(stack)-1]
            if s[i] == ')' && lastChar != '('{
                return false
            }
            if s[i] == '}' && lastChar != '{'{
                return false
            }
            if s[i] == ']' && lastChar != '['{
                return false
            }
        }
    }
    if len(stack) > 0{
        return false
    }
    return true
}

File: valid-parentheses/valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        CloseToOpen = {"]":"[", "}":"{", ")":"("}
        for r in s:
            if r in CloseToOpen:
                if stack and stack[-1] == CloseToOpen[r]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(r)
        return True if not stack else False
            
        